pages/visualizations/chaoss/contributors_over_time.py

```diff
@@ -139,64 +138,88 @@ def create_graph(repolist, timer_pings, contribs, interval):
     logging.debug("CONTRIBUTIONS_OVER_TIME_VIZ - START")
     start = time.perf_counter()
 
-    df_cont = pd.DataFrame(results)
-    df_cont["created_at"] = pd.to_datetime(df_cont["created_at"], utc=True, format="%Y-%m-%d")
+    # create dataframe from record data
+    df = pd.DataFrame(results)
+
+    # test if there is data
+    if df.empty:
+        logging.debug("PULL REQUESTS OVER TIME - NO DATA AVAILABLE")
+        return nodata_graph, False, dash.no_update
+
+    # convert to datetime objects with consistent column name
+    df["created_at"] = pd.to_datetime(df["created_at"], utc=True)
+    df.rename(columns={"created_at": "created"}, inplace=True)
+
+    # remove null contrib ids
+    df.dropna(inplace=True)
 
     # create column for identifying Drive by and Repeat Contributors
-    contributors = df_cont["cntrb_id"][df_cont["rank"] == contribs].to_list()
-    df_cont["type"] = np.where(df_cont["cntrb_id"].isin(contributors), "Repeat", "Drive-By")
+    contributors = df["cntrb_id"][df["rank"] == contribs].to_list()
 
-    # reset index to be ready for plotly
-    df_cont = df_cont.reset_index()
+    # dfs for drive by and repeat contributors
+    df_drive_temp = df.loc[~df["cntrb_id"].isin(contributors)]
+    df_repeat_temp = df.loc[df["cntrb_id"].isin(contributors)]
 
-    # time values for graph
-    x_r, x_name, hover, period = get_graph_time_values(interval)
+    # order values chronologically by creation date
+    df = df.sort_values(by="created", axis=0, ascending=True)
 
-    # graphs generated for aggregation by time interval
-    drive_temp = (
-        df_cont[df_cont["type"] == "Drive-By"]
-        .groupby(by=df_cont.created_at.dt.to_period(period))["cntrb_id"]
+    # variable to slice on to handle weekly period edge case
+    period_slice = None
+    if interval == "W":
+        period_slice = 10
```

Comment:
why the '10'? could you please comment that?

Reply:
its to handle the weekly period edge case. So when you do weekly it gives the period with the start and end state instead of just a single date like the others. You have to cut it off to convert from a period type to a datetime